### DISCUSSION TITLE
Prisma migration drift on self relating table relation column removal

### DISCUSSION THREAD

**Discussion by shokhboz-abdullaev** [prisma/prisma#21117] [Q&A]
### Question

Hello everyone, I'm facing problems while migrating. So to begin with, last month, we, in our team, had a DB restructuring for one of our app module. There we replaced same table relationship to (Cat model <-> Cat model) to a third party table relationship (Cat model -> Dog model <- Cat model). But we decided to keep the foreign key columns as it was and we agreed to remove them when we finish importing existing data in production to a new db structure. So when the time came to remove the fields: Prisma says:
Drift detected: Your database schema is not in sync with your migration history.
[*] Changed the `Cat` table
  [-] Removed foreign key on columns (otherCatId)
  [+] Added foreign key on columns (otherCatId) (edit

**Comment by ludralph** [comment]
Hello @Full-Stack-Shokhboz-Abdullayev 👋 

Thank you for raising this question.

This error occurs when your database schema is out of sync with your migrations history and migrations table. In your case, it seems like the changes you made to the `Cat` table are causing this drift.

Can you take a look at fixing failed migrations with [`migrate diff and db execute commands`](https://www.prisma.io/docs/guides/migrate/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute)?

**Reply by shokhboz-abdullaev** [reply]
The problem is I do not know what to put for the --to and --from options when using migrate diff could pls help me with that?

**Reply by shokhboz-abdullaev** [reply]
+ the above thing works when using migrate deploy but errors only in development (migrate dev) if I have the data from prod/stage.

**Reply by ludralph** [reply]
Hi @Full-Stack-Shokhboz-Abdullayev 
For development environment, you can follow this guide to resolve schema drift 

- https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/troubleshooting-development#fixing-schema-drift-in-a-development-environment

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by shokhboz-abdullaev** [reply]
I have a dump from production to first test out locally the migration process and then after resolving the issues if any, I will push it to prod. So in my case, even though I'm working in development environment, I do not want to loose the data, so how should I fix that?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
